### PR TITLE
backport: bump boost min

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
See commit

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: doc/dependencies.md [hunk 1]
```diff
@@ -3,31 +3,31 @@ Dependencies
...
+| Qt | [5.15.11](https://download.qt.io/official_releases/qt/) | [5.11.3](https://github.com/bitcoin/bitcoin/pull/24132)         | No | |  |
+| SQLite | [3.32.1](https://sqlite.org/download.html) | [3.7.17](https://github.com/bitcoin/bitcoin/pull/19077)         |  |  |  |
+| XCB |  |                                                                 |  |  | [Yes](https://github.com/dashpay/dash/blob/develop/depends/packages/qt.mk) (Linux only) |
+| systemtap ([tracing](tracing.md))| [4.8](https://sourceware.org/ftp/systemtap/releases/) |                                                                 |  |  | |
+| xkbcommon |  |                                                                 |  |  | [Yes](https://github.com/dashpay/dash/blob/develop/depends/packages/qt.mk) (Linux only) |
+| ZeroMQ | [4.3.1](https://github.com/zeromq/libzmq/releases) | 4.0.0                                                           | No |  |  |
+| zlib |  |                                                                 |  |  | [Yes](https://github.com/dashpay/dash/blob/develop/depends/packages/qt.mk) |
```
Comment (UdjinM6) [inline doc/dependencies.md:30]: 29066: why messing with whitespaces/markdown table here?

Comment (PastaPastaPasta) [reply doc/dependencies.md:30]: cause clion f***s up..

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
debug build fails for me with
```
test/util/setup_common.cpp:444:9: error: use of undeclared identifier 'BOOST_ASSERT'
  444 |         BOOST_ASSERT(cbTx.has_value());
      |         ^
test/util/setup_common.cpp:447:13: error: use of undeclared identifier 'BOOST_ASSERT'
  447 |             BOOST_ASSERT(false);
      |             ^
test/util/setup_common.cpp:450:13: error: use of undeclared identifier 'BOOST_ASSERT'
  450 |             BOOST_ASSERT(false);
      |             ^
```

backporting 24395 (23a4d5d9) fixes it.

**Comment by UdjinM6** [comment]
Hmm... lots of failures in CI.. there must be smth else then. Also, it looks like 24395 caused high CPU usage https://github.com/bitcoin/bitcoin/issues/27586 and was reverted in 27724... so we need that too (or drop 24395 for now) 🙈

**Comment by UdjinM6** [comment]
let's try 27783 (887241a07d)?

**Review by knst — APPROVED** [review]
utACK c0154c0d8cfb4ca05b1953586b7ea5fd40b7ab7a

**Pull request merged**